### PR TITLE
ci/reduce-pr-runs

### LINKED ISSUE #47: Reduce CI runs
Currently the CI runs twice when a PR is open; once for the PR, once for the push to the branch. Continue to run the CI for branch pushes, but perform format and lint checks when a merge is attempted.

### PR DESCRIPTION
**Pull request by bengsparks**
Closes #47 - WIP PR for checking how the CI now works.

### CODE REVIEW

**Comment by bengsparks** [comment]
Please only approve this PR, do not merge. I require approval because I need to see what happens when I click merge myself.

**Review by RecurvedBow — APPROVED** [review]

**Pull request merged**
